src/curl.c

```diff
@@ -69,9 +74,113 @@ int swupd_curl_init(void)
 		return -1;
 	}
 
+#ifdef FALLBACK_CAPATHS
+	/* parse and initialize CA paths. */
+	{
+		char *str = strdup(FALLBACK_CAPATHS);
+		char *tok;
+		char *ctx;
+		struct stat st;
+		int i;
+		int sz = 1;
+
+		fallback_capaths = (char **)malloc(sz*sizeof(char *));
+		fallback_capaths[0] = NULL;
+		i = 1;
+
+		tok = strtok_r(str, ":", &ctx);
+		do {
+			if (stat(tok, &st)) {
+				continue;
+			}
+			if ((st.st_mode & S_IFMT) != S_IFDIR) {
+				continue;
+			}
+			if (i == sz) {
+				sz <<= 1;
+				fallback_capaths = (char **)realloc(fallback_capaths, sz*sizeof(char *));
+			}
+			fallback_capaths[i] = strdup(tok);
+			i++;
+
+		} while ((tok = strtok_r(NULL, ":", &ctx)));
+		fallback_capaths_no = i;
+		free(str);
+	}
+#else
+	fallback_capaths = (char **)malloc(sizeof(char *));
+	fallback_capaths[0] = NULL;
+	fallback_capaths_no = 1;
+#endif
+
 	return 0;
 }
 
+static void swupd_curl_test_resume(void);
+
+int swupd_curl_check_network(void) {
+	CURLcode curl_ret;
+	int ret = -1;
+	int i;
+	CURL *c;
+	static int has_network = 0;
+
+	if (has_network) return 0;
```

Comment:
the `return 0;` should go on a newline and surrounded by brackets even though it is a single line for consistency.